version: '3.8'

services:
  pg:
    image: postgres:14.0-alpine
    restart: always
    volumes:
      - pg_data:/var/lib/postgresql/data
    environment:
      POSTGRES_USER: ${PG_USER}
      POSTGRES_PASSWORD: ${PG_PASSWORD}
      POSTGRES_DB: ${PG_DB}
    ports:
      - ${PG_PORT}:${PG_PORT}
    healthcheck:
      test: ['CMD', 'pg_isready', '-U', $PG_USER, '-d', $PG_DB]
      interval: 10s
      timeout: 5s
      retries: 5

  api:
    build:
      context: .
      dockerfile: src/config/Dockerfile.dev
    restart: always
    volumes:
      - .:/app
    environment:
      PG_USER: ${PG_USER}
      PG_PASSWORD: ${PG_PASSWORD}
      PG_DB: ${PG_DB}
      PG_PORT: ${PG_PORT}
      PG_HOST: ${PG_HOST}
      PORT: ${PORT}
    ports:
      - ${PORT}:${PORT}
    entrypoint: ['/bin/sh', './entrypoint.sh']
    depends_on:
      pg:
        condition: service_healthy

volumes:
  pg_data: {}
